boot/bootutil/src/image_ec256.c

```diff
@@ -178,6 +230,13 @@ bootutil_verify_sig(uint8_t *hash, uint32_t hlen, uint8_t *sig, size_t slen,
     /*
      * This is simplified, as the hash length is also 32 bytes.
      */
+#ifdef CY_MBEDTLS_HW_ACCELERATION
```

Comment:
Can we do this without the ifdefs? Or at least can we do it so that the ifdefs are in the header file instead of here.

Although, I am kind of realizing that the API surface isn't quite right to make this clean, and it just gets a bit messier each time we add one. Right now, though, having most of the ifdef blocks in `ecdsa_p256.h` and a few here isn't ideal.
The difference seems to be in terms of how a given backend wants the keys formatted. I suspect the key decoding needs to be what is conditionalized, but these are large enough that it doesn't really make sense to put it in the header file. In fact, there isn't really any benefit to putting the ifdefs in the header file (or even having it), since it is only included by a single C file.